src/filter_tree/ft_to_rsq.c

```diff
@@ -546,10 +546,8 @@ RSQNode *FilterTreeToQueryNode
 	ASSERT(tree != NULL);
 	ASSERT(none_converted_filters != NULL);
 
-	// clone filter tree, as it is about to be modified
-	FT_FilterNode  *t       =  FilterTree_Clone(tree);
```

Comment:
Not sure if this is OK

Consider `op_node_by_index_scan` which passes a `filter tree` to this function and `none_converted_filters` is populated with sub trees from the original `filter tree`.

Now when `op_node_by_index_scan` gets reseted `IndexScanReset` will free `op->unresolved_filters` which will damage the original `filter tree`.

I'm thinking about a `cartesian product` which utilize the `IndexScanOp` operation but the filter has unresolved filters